File: 389-find-the-difference/find-the-difference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        # result = [x for x in t if x not in s]  --> Fails when the extra character already exists in 
        # result_str = ""                           s but just has higher frequency.
        # for i in result:
        #     result_str = result_str + str(i)
        # return result_str
        result = ""
        for ch in t:
            if s.count(ch) != t.count(ch):
                result = ch
                break
        return result

File: 389-find-the-difference/find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        HashMap<Character, Integer> mp = new HashMap<>();

        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            mp.put(c, mp.getOrDefault(c, 0)+1);
        }

        for(int i=0; i<t.length(); i++){
            char c = t.charAt(i);
            mp.put(c, mp.getOrDefault(c, 0)-1);
        }

        for(char key : mp.keySet()){
            if(mp.get(key) < 0){
                return key;
            }
        }

        return '1';
    }
}